### PR TITLE
Add sort field configuration and fix config related issues

### PR DESCRIPTION
**Pull request by fmassot**
Fixes:
- [x] #486 
- [x] #565 
- [x] #564
- [x] #360

### CODE REVIEW

File: quickwit-index-config/src/default_index_config/default_config.rs [hunk 1]
```diff
@@ -43,20 +43,36 @@ pub struct DefaultIndexConfigBuilder {
...
+}
+
+impl From<SortByConfig> for SortBy {
+    fn from(sort_by_config: SortByConfig) -> Self {
+        SortBy::SortByFastField {
+            field_name: sort_by_config.field_name,
+            order: sort_by_config.order,
+        }
+    }
+}
```
Comment (PSeitz) [inline quickwit-index-config/src/default_index_config/default_config.rs:66]: looks like SortBy::SortByFastField is the same as SortByConfig, so we could reuse it there
```
#[derive(Clone, Debug)]
pub enum SortBy {
    /// Sort by a specific field.
    SortByFastField (SortByFastFieldConfig),
    /// Sort by DocId
    DocId,
}
```

File: quickwit-indexing/src/actors/indexer.rs [hunk 2]
```diff
@@ -93,8 +93,24 @@ enum PrepareDocumentOutcome {
...
     fn create_indexed_split(&self) -> anyhow::Result<IndexedSplit> {
         let schema = self.index_config.schema();
+        let mut index_settings = IndexSettings::default();
+        let sort_by_field = match self.index_config.sort_by() {
+            SortBy::DocId => None,
+            SortBy::SortByFastField { field_name, order } => {
+                let tantivy_order = match order {
+                    SortOrder::Asc => Order::Asc,
+                    SortOrder::Desc => Order::Desc,
+                };
```
Comment (PSeitz) [inline quickwit-indexing/src/actors/indexer.rs:103]: replace with into?

**Review by PSeitz — APPROVED** [review]
LGTM except minor comments

File: quickwit-index-config/src/default_index_config/default_config.rs (hunk 1 above)
Comment (fulmicoton) [reply quickwit-index-config/src/default_index_config/default_config.rs:66]: I prefer the way it is today.

File: quickwit-index-config/src/default_index_config/default_config.rs [hunk 3]
```diff
@@ -87,26 +103,50 @@ impl DefaultIndexConfigBuilder {
...
                         timestamp_field_name
                     )
                 }
-            } else {
+            }
+        }
+
+        // Resolve sort field
+        let mut sort_by = None;
+        if let Some(sort_by_config) = self.sort_by {
```
Comment (fulmicoton) [inline quickwit-index-config/src/default_index_config/default_config.rs:133]: can we remove the `mut` ?

Either by using map, or by extracting the conversion in a function.

let sort_by = resolve_sort_field(...);

File: quickwit-index-config/src/default_index_config/default_config.rs (hunk 1 above)
Comment (fmassot) [reply quickwit-index-config/src/default_index_config/default_config.rs:66]: @PSeitz this is the way :D 
I was hesitating and have found that it was slightly hurting readability and I like keeping away all the bloated config structs away from the core as much as possible. I would keep it as it is this for now.

**Pull request merged**